pkg/mcp/snapshot/snapshot.go

```diff
@@ -43,15 +43,17 @@ type Cache struct {
 	groupIndex GroupIndexFn
 }
 
-// GroupIndexFn returns a stable group index for the given MCP node.
-type GroupIndexFn func(node *mcp.SinkNode) string
+// GroupIndexFn returns a stable group index for the given MCP collection and node.
+type GroupIndexFn func(collection string, node *mcp.SinkNode) string
 
 // DefaultGroup is the default group when using the DefaultGroupIndex() function.
 const DefaultGroup = "default"
 
+var _ GroupIndexFn = DefaultGroupIndex
+
 // DefaultGroupIndex provides a default GroupIndexFn function that
 // is usable for testing and simple deployments.
-func DefaultGroupIndex(_ *mcp.SinkNode) string {
+func DefaultGroupIndex(_ string, _ *mcp.SinkNode) string {
```

Comment:
Should we even have this anymore? The implementation in Galley is the actual convention of the MCP protocol, for all intents and purposes. I think we should either capture that here, or simply get rid of this to avoid confusion. @ayj

Reply:
The group index idea was explicitly designed so that MCP doesn't need to care about how a particular collection's requests map onto groups. IMO, that is a responsibility of the runtime layer's publishing strategy. I think it's worthwhile to provide some useful out-of-the-box for testing and simple cases. We could put it in another file/package to make this clearer.

Reply:
@ayj would examples in the mcp testing package be sufficient? I think implementing a one-line index function is a pretty low barrier to entry.

Reply:
Sure.